components/NewRepoModal.js

```diff
@@ -6,13 +6,22 @@ import {
   StyleSheet,
   TouchableOpacity,
 } from 'react-native';
+import { connect } from 'react-redux';
 import { Actions } from 'react-native-router-flux';
+import { AddItemAction } from '../actions'
 
-export default class NewRepoModal extends Component {
+class NewRepoModal extends Component {
   state = {
     modalText: '',
   };
 
+  gettingApi = async () => {
+    const call = await fetch(`http://api.github.com/repos/${this.state.modalText}`);
+    const response = await call.json();
+    this.props.addItemAction(this.state.modalText, response);
+    Actions.pop()
+  }
+
   onChangeText = (modalText) => this.setState({ modalText });
```

Comment:
Since you started using redux, you can move this method that changes the `modalText` to redux state, with its own action creator and etc. Then you can remove the local state of this component.